### PR TITLE
[ci] Move CSCS tests and configuration out of this repo

### PR DESCRIPTION
**Pull request by vkarak**
The CSCS tests and the CSCS site-specific configuration have been moved to https://github.com/eth-cscs/cscs-reframe-tests.

There are still a couple of subtasks that need to be addressed before we merge this PR (see #2511 for more details).

Addresses #2511.

### CODE REVIEW

**Review by victorusu — APPROVED** [review]
`lgtm`

File: docs/tutorial_advanced.rst [hunk 1]
```diff
@@ -681,7 +681,7 @@ First, we need to enable the container platform support in ReFrame's configurati
    :emphasize-lines: 15-24
 
 For each partition, users can define a list of container platforms supported using the :js:attr:`container_platforms` `configuration parameter <config_reference.html#.systems[].partitions[].container_platforms>`__.
-In this case, we define the `Sarus <https://github.com/eth-cscs/sarus>`__ platform for which we set the :js:attr:`modules` parameter in order to instruct ReFrame to load the ``sarus`` module, whenever it needs to run with this container platform.
+In this case, we define the `Sarus <https://github.com/reframe-hpc/sarus>`__ platform for which we set the :js:attr:`modules` parameter in order to instruct ReFrame to load the ``sarus`` module, whenever it needs to run with this container platform.
```
Comment (ekouts) [inline docs/tutorial_advanced.rst:684]: This should remain eth-cscs

**Review by ekouts — COMMENTED** [review]
Is `config/cscs-ci.py` on the repo left on purpose?

File: docs/tutorial_advanced.rst (hunk 1 above)
Comment (vkarak) [reply docs/tutorial_advanced.rst:684]: Correct 😅

**Comment by vkarak** [comment]
> Is config/cscs-ci.py on the repo left on purpose?

Yes, because it's needed for the CSCS CI that tests the scheduler backends.

**Review by ekouts — COMMENTED** [review]
This is not necessary for this PR but since I saw that you changes the links to `reframe-hpc`, I think we also need to change the repo to reframe-hpc in the following files:
- `.pep8speaks.yml`
- `CONTRIBUTING.md`
- `setup.cfg`
- `ci-scripts/deploy.sh`
- `docs/conf.py`
- `hpctestlib/microbenchmarks/gpu/gpu_burn.py`
- `reframe/frontend/cli.py`
- `reframe/schemas/config.json`
- `reframe/schemas/runreport.json`
- `reframe/utility/osext.py`
- `unittests/test_config.py`
- `unittests/test_utility.py`
There are links to the contributing guidelines, to old issues etc..

**Comment by vkarak** [comment]
@ekouts That's a valid observation. I will fix those links.

**Review by ekouts — APPROVED** [review]

**Pull request merged**